src/Controller/Component.php

```diff
@@ -170,7 +170,7 @@ public function implementedEvents(): array
             'Controller.startup' => 'startup',
             'Controller.beforeRender' => 'beforeRender',
             'Controller.beforeRedirect' => 'beforeRedirect',
-            'Controller.shutdown' => 'shutdown',
+            'Controller.shutdown' => 'afterFilter',
```

Comment:
Deprecating this somehow would be nice. It's a fairly hidden change that users won't understand.

We'd have to implement `afterFilter()` and the default implementation would check for the existance of `shutdown()`.

Reply:
Yeah, we can deprecate in 4.3 I guess.

Reply:
I don't think the callback would be used often. We can add a `Component::shutdown()` in 5.0 itself which throws a message telling using to use `Component::afterFilter()` instead.

Reply:
True. But https://github.com/cakephp/cakephp/pull/15887 seems cleaner as we then can remove the baggage in 5.0 directly, not having to continue the fallback until 6.x